docs/dev/building/Compile.rst

```diff
@@ -0,0 +1,456 @@
+  have checked out the master branch of some submodules, running ``git pull``
+  in those submodules is also an option.)
+
+  Rarely, we add or remove submodules. If there are any changes to the existence
+  of submodules when you switch between branches, you should run
+  ``git submodule update --init`` instead (adding ``--init`` to the above
+  command).
+
+  Some common errors that can arise when failing to update submodules include:
+
+  * ``fatal: <some path> does not exist`` when performing Git operations
+  * Build errors, particularly referring to structures in the ``df::`` namespace
+    or the ``library/include/df`` folder
+  * ``Not a known DF version`` when starting DF
+  * ``Run 'git submodule update --init'`` when running CMake
+
+  Submodules are a particularly confusing feature of Git. The
+  `Git Book <https://git-scm.com/book/en/v2/Git-Tools-Submodules>`_ has a
+  thorough explanation of them (as well as of many other aspects of Git) and
+  is a recommended resource if you run into any issues. Other DFHack developers
+  are also able to help with any submodule-related (or Git-related) issues
+  you may encounter.
+
+Dependencies
+------------
+
+If you haven't already checked, this would be a good point to ensure you have
+all the `dependencies <build-dependencies>`.
+
+
+Contributing to DFHack
+----------------------
+
+For details on contributing to DFHack, including pull requests, code
+format, and more, please see `contributing-code`.
+
+
+Build settings
+==============
+
+This section describes build configuration options that apply to all platforms.
+If you don't have a working build environment set up yet, follow the instructions
+in the platform-specific sections below first, then come back here.
+
+Be sure to check out common `build options <build-options>`.
+
+Instructions
+============
+
+.. contents::
+  :local:
+  :depth: 1
+
+.. _compile-linux:
+
+Linux
+-----
+On Linux, DFHack acts as a library that shadows parts of the SDL API using LD_PRELOAD.
+
+Build
+~~~~~
+Building is fairly straightforward. Enter the ``build`` folder (or create an
+empty folder in the DFHack directory to use instead) and start the build like this::
+
+    cd build
+    cmake .. -G Ninja -DCMAKE_BUILD_TYPE:string=Release -DCMAKE_INSTALL_PREFIX=<path to DF>
+    ninja install  # or ninja -jX install to specify the number of cores (X) to use
+
+<path to DF> should be a path to a copy of Dwarf Fortress, of the appropriate
+version for the DFHack you are building. This will build the library along
+with the normal set of plugins and install them into your DF folder.
+
+Alternatively, you can use ccmake instead of cmake::
+
+    cd build
+    ccmake .. -G Ninja
+    ninja install
+
+This will show a curses-based interface that lets you set all of the
+extra options. You can also use a cmake-friendly IDE like KDevelop 4
+or the cmake-gui program.
+
+Windows cross compiling from Linux
```

Comment:
I think this should be on the same header level as the main operating systems, since it is a significantly different use case

Reply:
How about under the heading: Docker?
Cause otherwise I have the differing opinion that this is the best hierarchy.

Reply:
I think I'd be happy as long as it shows up in the TOC on this page.

Reply:
My vote is for "Windows cross-compiling from Linux". No opinion on the hierarchy (nested or sibling with "Windows" are fine with me).

Reply:
I need a tie breaker

Reply:
Ok, my vote is "if it ain't broke, don't fix it". So `Docker` => `Windows cross compiling from Linux`. The instructions only apply to building for Windows currently, so `Docker` is ambiguous.